src/DearImGui.hs

```diff
@@ -842,4 +842,100 @@ setNextWindowCollapsed b (ImGuiCond con) = liftIO do
 -- Wraps @ImGui::SetNextWindowBgAlpha()@
 setNextWindowBgAlpha :: (MonadIO m) => CFloat -> m ()
 setNextWindowBgAlpha f = liftIO do
-  [C.exp| void { SetNextWindowBgAlpha($(float f)) } |]
\ No newline at end of file
+  [C.exp| void { SetNextWindowBgAlpha($(float f)) } |]
+
+-- | undo a sameLine or force a new line when in an horizontal-layout context.
+--
+-- Wraps @ImGui::NewLine()@
+newLine :: (MonadIO m) => m ()
+newLine = liftIO do
+  [C.exp| void { NewLine() } |]
+
+-- | Add vertical spacing.
+--
+-- Wraps @ImGui::Spacing()@
+spacing :: (MonadIO m) => m ()
+spacing = liftIO do
+  [C.exp| void { Spacing() } |]
+
+-- | Add a dummy item of given size. unlike `invisibleButton`, `dummy` won't take the mouse click or be navigable into.
+--
+-- Wraps @ImGui::Dummy()@
+dummy :: (MonadIO m, HasGetter ref ImVec2) => ref -> m ()
+dummy sizeRef = liftIO do 
+  size' <- get sizeRef
+  with size' $ \ sizePtr -> [C.exp| void { Dummy(*$(ImVec2 *sizePtr)) } |]
+
+-- | Move content position toward the right, by indent_w, or style.IndentSpacing if indent_w <= 0
+--
+-- Wraps @ImGui::Indent()@
+indent :: (MonadIO m) => CFloat -> m ()
```

Comment:
I think we should use `Float` instead of `CFloat` in the user interface; at least that's what @ocharles has been doing so far.

Reply:
can do